install/tools/ipactl

```diff
@@ -134,16 +135,25 @@ def emit_err(err):
     sys.stderr.write(err + '\n')
 
 
-def version_check():
+def version_check(options):
     try:
         installutils.check_version()
     except (installutils.UpgradeMissingVersionError,
             installutils.UpgradeDataOlderVersionError) as exc:
         emit_err("IPA version error: %s" % exc)
-        emit_err("Upgrade required: please run ipa-server-upgrade command")
-        raise IpactlError("Aborting ipactl")
     except installutils.UpgradeVersionError as e:
         emit_err("IPA version error: %s" % e)
+    else:
+        return
+
+    emit_err("Automatically running upgrade, for details see {}".format(
+             paths.IPAUPGRADE_LOG))
+
+    # Fork out to call ipa-server-upgrade so that logging is sane.
+    result = run([paths.IPA_SERVER_UPGRADE], raiseonerr=False,
+                  capture_error=True)
+    if result.returncode != 0:
+        emit_err("Automatic upgrade failed: %s" % result.error_output)
```

Comment:
Maybe we can instruct the user to try and run `ipa-server-upgrade` manually?

Reply:
Usually, when there's something which takes a bit time, we tend to tell the user. Here, we're sitting for about 5 minutes without output. I believe you should consider my previous comment and also add a warning that the upgrade process may take some time.